docs/using-the-compiler.rst

```diff
@@ -182,6 +182,7 @@ at each version. Backward compatibility is not guaranteed between each version.
    - Opcode ``mcopy`` is available in assembly (see `EIP-5656 <https://eips.ethereum.org/EIPS/eip-5656>`_).
    - Opcodes ``tstore`` and ``tload`` are available in assembly (see `EIP-1153 <https://eips.ethereum.org/EIPS/eip-1153>`_).
 - ``prague`` (**experimental**)
+- ``osaka``
```

Comment:
I guess this should be marked experimental?

Reply:
It definitely should.

Also, one thing I noticed recently, that we might want to fix (though maybe in a separate PR) is that we do have an `experimental` flag in metadata, but we do not set it for experimental EVM versions. So bytecode compiled for Prague is not currently marked as experimental.

Reply:
It's marked as experimental in the other PR (the one that moves EOF tests to Osaka) - https://github.com/ethereum/solidity/pull/15828/files#diff-c907e10ad187a2840ede6859989664668693753e8c7d860fdc1b60275e39f053R185

Not sure why it was done there; I also have a sneaking suspicion that this PR will start failing as soon as this PR and the linked one above are rebased, since Kamil said that we've introduced semantic tests for EOF, whereas evmone 13 (latest and now merged) release did not move the EOF implementation to Osaka.

Reply:
On Monday: #15666.

Reply:
> On Monday: #15666.

Ah, so this doesn't actually introduce any semantic tests, just the support for the eventual addition of such tests. I guess in that case it doesn't matter as much, but if we go forward with the "move EOF from Prague to Osaka" before we bump images to use the yet unreleased evmone, such tests will not be able to be run.

Reply:
> Ah, so this doesn't actually introduce any semantic tests, just the support for the eventual addition of such tests. 

#15665 does introduce a few new semantic tests. But we don't really have to introduce new tests to have them. We have plenty already and almost all of them now run on EOF as well (#15552).

> such tests will not be able to be run.

Why wouldn't they be able to run? They are not marked as requiring Prague specifically.

Reply:
Yeah, I've pushed the requested changes already. Can you take another look, and if it's OK, I'll merge it later today (have to leave now).

Reply:
BTW, I also added an issue to fix the missing experimental flag: #15835.